core/eth/reader.go

```diff
@@ -616,42 +618,119 @@ func (t *Reader) GetAllVersionedBlobParams(ctx context.Context) (map[uint8]*core
 	return res, nil
 }
 
-func (t *Reader) GetActiveReservations(ctx context.Context, blockNumber uint32, accountIDs []string) (map[string]core.ActiveReservation, error) {
-	// contract is not implemented yet
-	return map[string]core.ActiveReservation{}, nil
+func (t *Reader) GetActiveReservations(ctx context.Context, accountIDs []string) (map[string]core.ActiveReservation, error) {
+	// map accountIDs to addresses
+	accountAddresses := make([]gethcommon.Address, len(accountIDs))
+	for i, accountID := range accountIDs {
+		accountAddresses[i] = gethcommon.HexToAddress(accountID)
+	}
+
+	reservations_map := make(map[string]core.ActiveReservation)
+	reservations, err := t.bindings.PaymentVault.GetReservations(&bind.CallOpts{
+		Context: ctx,
+	}, accountAddresses)
+	if err != nil {
+		return nil, err
+	}
+
+	// since reservations are returned in the same order as the accountIDs, we can directly map them
+	for i, reservation := range reservations {
+		reservations_map[accountIDs[i]] = reservation
```

Comment:
Is it guaranteed that `PaymentVault.GetReservations` returns reservations in the same order?
validate that `len(accountIDs) = len(reservations)` before this loop?

Reply:
it looks like the ordering and length is guaranteed by [this loop](https://github.com/Layr-Labs/eigenda/pull/800/files#diff-2f1b4eb845063422ced2032eace4bbe25fc506e57398f8ed8d1f0fa00a971b4eR131). If there's no match in the mapping, as you commented, would be a zero value. I will update it to set a nil or empty reservation when there's a zero value, do you think that's sufficient?